visual/common.go

```diff
@@ -36,3 +45,13 @@ func Box(w io.Writer, name string, label string) {
 		`%s [label="%s" shape=box margin=0.1 width=0 height=0 fillcolor="#FF6403" style=filled];
 `, name, label)
 }
+
+func Cluster(w io.Writer, number int, label string, display string) {
```

Comment:
You can split it into two functions - `StartCluster` and `EndCluster` so that you can write directly into the file instead of accumulating `clusterData` first